### ISSUE TITLE
self-assigning-variable false positive on imported name

### ISSUE THREAD

**Issue by scop** [PyCQA/pylint#3151]
pylint 2.4.2, astroid 2.3.1

```
$ cat t.py
from socket import AF_UNIX
AF_UNIX = AF_UNIX
$ pylint t.py | grep self-assign
t.py:2:0: W0127: Assigning the same variable 'AF_UNIX' to itself (self-assigning-variable)
```

**Comment by scop** [comment]
Hm, on a 2nd thought, maybe not a bug after all, but leaving open for second opinions.

**Comment by PCManticore** [comment]
Yeah, this check should catch situations such as these, where the original intent might have been to assign the value to a different name or to an attribute bearing the same name.

**Comment by scop** [comment]
FWIW the case where I saw these was importing a constant from module A under the same name in B, with the intent that it constitutes an API constant in B and should be accessed like B.CONSTANT when dealing with things related to B. But the assignment is superfluous as the import already provides for that (the visibility). Removal of this self-assignment might now require disabling an unused import warning though in case the constant is not being used in code within B itself.
